### PR TITLE
fix(mgmt_cli_test): address test_client_encryption flakiness

### LINKED ISSUE #6436: manager-3.1: test_client_encryption is flakky
### Versions
* **SCT**: manager-3.1

### Logs
Example of failure:
```
[2023-07-28T23:12:55.891Z] 2023-07-28 23:12:45.579: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=d0152a39-ee24-4273-8ed5-7b9c3a2bbc99, source=MgmtCliTest.test_manager_sanity (mgmt_cli_test.MgmtCliTest) [Client Encryption]() message=Traceback (most recent call last):
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 69, in wait_for
[2023-07-28T23:12:55.892Z] res = retry(func, **kwargs)
[2023-07-28T23:12:55.892Z] File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 404, in __call__
[2023-07-28T23:12:55.892Z] do = self.iter(retry_state=retry_state)
[2023-07-28T23:12:55.892Z] File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 360, in iter
[2023-07-28T23:12:55.892Z] raise retry_exc.reraise()
[2023-07-28T23:12:55.892Z] File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 194, in reraise
[2023-07-28T23:12:55.892Z] raise self
[2023-07-28T23:12:55.892Z] tenacity.RetryError: RetryError[<Future at 0x7fe558679bd0 state=finished returned bool>]
[2023-07-28T23:12:55.892Z] The above exception was the direct cause of the following exception:
[2023-07-28T23:12:55.892Z] Traceback (most recent call last):
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/mgmt/cli.py", line 399, in wait_for_status
[2023-07-28T23:12:55.892Z] return wait.wait_for(func=self.is_status_in_list, step=step, throw_exc=True,
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 80, in wait_for
[2023-07-28T23:12:55.892Z] raise WaitForTimeoutError(err) from ex
[2023-07-28T23:12:55.892Z] sdcm.wait.WaitForTimeoutError: Wait for: Waiting until task: repair/e00e1ee8-1a27-4a49-8a68-b5d19d21287e reaches status of: ['ERROR', 'ERROR (4/4)']: timeout - 240 seconds - expired
[2023-07-28T23:12:55.892Z] The above exception was the direct cause of the following exception:
[2023-07-28T23:12:55.892Z] Traceback (most recent call last):
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/mgmt_cli_test.py", line 377, in test_manager_sanity
[2023-07-28T23:12:55.892Z] self.test_client_encryption()
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/mgmt_cli_test.py", line 714, in test_client_encryption
[2023-07-28T23:12:55.892Z] repair_task.wait_for_status(list_status=[TaskStatus.ERROR, TaskStatus.ERROR_FINAL], step=5, timeout=240)
[2023-07-28T23:12:55.892Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/mgmt/cli.py", line 403, in wait_for_status
[2023-07-28T23:12:55.892Z] raise WaitForTimeoutError(
[2023-07-28T23:12:55.892Z] sdcm.wait.WaitForTimeoutError: Failed on waiting until task: repair/e00e1ee8-1a27-4a49-8a68-b5d19d21287e reaches status of ['ERROR', 'ERROR (4/4)']: current task status DONE: Wait for: Waiting until task: repair/e00e1ee8-1a27-4a49-8a68-b5d19d21287e reaches status of: ['ERROR', 'ERROR (4/4)']: timeout - 240 seconds - expired
```

### Failure N1
[job](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/centos-sanity-ipv6-test/650)
[db-cluster](https://cloudius-jenkins-test.s3.amazonaws.com/d79aa866-f31d-4766-8d30-0c3fad6f3c50/20230728_231307/db-cluster-d79aa866.tar.gz)
[sct-runner-events](https://cloudius-jenkins-test.s3.amazonaws.com/d79aa866-f31d-4766-8d30-0c3fad6f3c50/20230728_231307/sct-runner-events-d79aa866.tar.gz)
[sct-runner-python-log](https://cloudius-jenkins-test.s3.amazonaws.com/d79aa866-f31d-4766-8d30-0c3fad6f3c50/20230728_231307/sct-d79aa866.log.tar.gz)
[loader-set](https://cloudius-jenkins-test.s3.amazonaws.com/d79aa866-f31d-4766-8d30-0c3fad6f3c50/20230728_231307/loader-set-d79aa866.tar.gz)
[monitor-set](https://cloudius-jenkins-test.s3.amazonaws.com/d79aa866-f31d-4766-8d30-0c3fad6f3c50/20230728_231307/monitor-set-d79aa866.tar.gz)

### Failure N2
[job](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/centos-sanity-ipv6-test/649)
[db-cluster](https://cloudius-jenkins-test.s3.amazonaws.com/581af535-0cec-498e-8547-1eea318e4b86/20230727_225019/db-cluster-581af535.tar.gz)
[sct-runner-events](https://cloudius-jenkins-test.s3.amazonaws.com/581af535-0cec-498e-8547-1eea318e4b86/20230727_225019/sct-runner-events-581af535.tar.gz)
[sct-runner-python-log](https://cloudius-jenkins-test.s3.amazonaws.com/581af535-0cec-498e-8547-1eea318e4b86/20230727_225019/sct-581af535.log.tar.gz)
[loader-set](https://cloudius-jenkins-test.s3.amazonaws.com/581af535-0cec-498e-8547-1eea318e4b86/20230727_225019/loader-set-581af535.tar.gz)
[monitor-set](https://cloudius-jenkins-test.s3.amazonaws.com/581af535-0cec-498e-8547-1eea318e4b86/20230727_225019/monitor-set-581af535.tar.gz)

### Failure N3
[job](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/centos-sanity-ipv6-test/648)
[db-cluster](https://cloudius-jenkins-test.s3.amazonaws.com/4b18fe08-2514-4874-b804-890f3c92e469/20230726_233639/db-cluster-4b18fe08.tar.gz)
[sct-runner-events](https://cloudius-jenkins-test.s3.amazonaws.com/4b18fe08-2514-4874-b804-890f3c92e469/20230726_233639/sct-runner-events-4b18fe08.tar.gz)
[sct-runner-python-log](https://cloudius-jenkins-test.s3.amazonaws.com/4b18fe08-2514-4874-b804-890f3c92e469/20230726_233639/sct-4b18fe08.log.tar.gz)
[loader-set](https://cloudius-jenkins-test.s3.amazonaws.com/4b18fe08-2514-4874-b804-890f3c92e469/20230726_233639/loader-set-4b18fe08.tar.gz)
[monitor-set](https://cloudius-jenkins-test.s3.amazonaws.com/4b18fe08-2514-4874-b804-890f3c92e469/20230726_233639/monitor-set-4b18fe08.tar.gz)

### Description
Test creates a repair task, does something then enables client ssl, restart scylla nodes and assert repair status is an ERROR.
Since there is a gap between creating a repair task and restarting scylla, it is possible that task is finished before scylla is restarted.
As result test can falsely fail.
 
https://github.com/scylladb/scylla-cluster-tests/blob/c32d1092169479ce5a4a80508b8655b3c7f9846e/mgmt_cli_test.py#L702-L723

### PR DESCRIPTION
**Pull request by dkropachev**
Closes #6436 

Test creates a repair task, does something then enables client ssl, restart scylla nodes and assert repair status is an ERROR.
Since there is a gap between creating a repair task and restarting scylla, it is possible that task is finished before scylla is restarted.
As result test can falsely fail.

Solution - to move repair task creating after scylla is being reconfigured

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/7

**Review by fgelcer — DISMISSED** [review]
LGTM

**Comment by fgelcer** [comment]
> Test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/7

it failed... seems like it wasn't enough

File: mgmt_cli_test.py [hunk 1]
```diff
@@ -710,7 +709,7 @@ def test_client_encryption(self):
                 DbEventsFilter(db_event=DatabaseLogEvent.RUNTIME_ERROR, line="get_repair_meta: repair_meta_id"):
 
             self.db_cluster.enable_client_encrypt()
-
+            repair_task = mgr_cluster.create_repair_task(fail_fast=True)
             repair_task.wait_for_status(list_status=[TaskStatus.ERROR, TaskStatus.ERROR_FINAL], step=5, timeout=240)
```
Comment (fruch) [inline mgmt_cli_test.py:713]: task is waited to be error, cause it change from non encryption to encryption.

Comment (dkropachev) [reply mgmt_cli_test.py]: Exactly, I am not expecting it to succeed, it is by design.

**Comment by dkropachev** [comment]
> > Test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/7
> 
> it failed... seems like it wasn't enough

It is failed because there is some problem is `s-m` that does not let it fail fast on client authentication error.

**Comment by fruch** [comment]
> > > Test run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/7
> > 
> > it failed... seems like it wasn't enough
> 
> It is failed because there is some problem is `s-m` that does not let it fail fast on client authentication error.
> 

Client encryption has nothing to do with repair.

The only reason the repair was failing was cause all the nodes were rebooted during the task.

**Comment by dkropachev** [comment]
I have re-run the test with increased timeout - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/8

**Comment by fgelcer** [comment]
> I have re-run the test with increased timeout - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/8

@dkropachev , this task that was expected to fail, still did not fail:
```
sdcm.wait.WaitForTimeoutError: Failed on waiting until task: repair/4a54e053-b57d-49fd-bc44-a117c6572e99 reaches status of ['ERROR', 'ERROR (4/4)']: current task status RUNNING: Wait for: Waiting until task: repair/4a54e053-b57d-49fd-bc44-a117c6572e99 reaches status of: ['ERROR', 'ERROR (4/4)']: timeout - 240 seconds - expired
```

**Comment by dkropachev** [comment]
> > I have re-run the test with increased timeout - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/8
> 
> @dkropachev , this task that was expected to fail, still did not fail:
> 
> ```
> sdcm.wait.WaitForTimeoutError: Failed on waiting until task: repair/4a54e053-b57d-49fd-bc44-a117c6572e99 reaches status of ['ERROR', 'ERROR (4/4)']: current task status RUNNING: Wait for: Waiting until task: repair/4a54e053-b57d-49fd-bc44-a117c6572e99 reaches status of: ['ERROR', 'ERROR (4/4)']: timeout - 240 seconds - expired
> ```

This is most-likely bug in `s-m`, i will check on it tomorrow - https://github.com/scylladb/scylla-manager/issues/3511

**Comment by dkropachev** [comment]
@fruch , @fgelcer , ok, I have updated test to handle wrong behavior from `s-m` and mark it to update it back when it is fixed

**Comment by dkropachev** [comment]
@fruch , updated test, run - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/11

**Comment by dkropachev** [comment]
@fruch , finally got it fixed, test run is here - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dkropachev/job/3-1-centos-sanity-test/20/

**Review by fruch — DISMISSED** [review]
LGTM

**Comment by fruch** [comment]
@ShlomiBalalis @fgelcer, can one of you review as well ?

**Review by fgelcer — DISMISSED** [review]
LGTM, but i want @ShlomiBalalis 's opinion, as he is much more familiar with this test than anyone else

**Review by ShlomiBalalis — APPROVED** [review]

**Pull request merged**